File: GXpert/GXpert.Web/Modules/LiveSessions/LiveSession/LiveSessionRow.cs
using GXpert.Web.Enums;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Serenity.Extensions.Entities;
using System;
using System.ComponentModel;

namespace GXpert.LiveSessions;

[ConnectionKey("Default"), Module("LiveSessions"), TableName("LiveSessions")]
[DisplayName("Live Session"), InstanceName("Live Session")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript("LiveSessions.LiveSession")]
public sealed class LiveSessionRow : LoggingRow<LiveSessionRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Meeting Id"), Size(2000), NotNull, QuickSearch, NameProperty]
    public string MeetingId { get => fields.MeetingId[this]; set => fields.MeetingId[this] = value; }
    [DisplayName("Title"), Size(255)]
    public string Title { get => fields.Title[this]; set => fields.Title[this] = value; }

    [DisplayName("Live Provider"), NotNull]
    public ELiveProvider? LiveProvider { get => (ELiveProvider?)fields.LiveProvider[this]; set => fields.LiveProvider[this] = (short?)value; }

    [DisplayName("Password"), Size(200), NotNull]
    public string Password { get => fields.Password[this]; set => fields.Password[this] = value; }

    [DisplayName("Secret"), Size(200), NotNull]
    public string Secret { get => fields.Secret[this]; set => fields.Secret[this] = value; }

    [DisplayName("Live Additional Info"), Size(2000), NotNull]
    public string LiveAdditionalInfo { get => fields.LiveAdditionalInfo[this]; set => fields.LiveAdditionalInfo[this] = value; }

    [DisplayName("Search Tags"), Size(1000)]
    public string SearchTags { get => fields.SearchTags[this]; set => fields.SearchTags[this] = value; }

    [DisplayName("Is Active"), DefaultValue(1)]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public StringField MeetingId;
        public Int16Field LiveProvider;
        public StringField Password;
        public StringField Secret;
        public StringField Title;
        public StringField LiveAdditionalInfo;
        public StringField SearchTags;
        public BooleanField IsActive;

    }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/LiveSessions/LiveSessionRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";
import { ELiveProvider } from "../Web/Enums.ELiveProvider";

export interface LiveSessionRow {
    Id?: number;
    MeetingId?: string;
    LiveProvider?: ELiveProvider;
    Password?: string;
    Secret?: string;
    Title?: string;
    LiveAdditionalInfo?: string;
    SearchTags?: string;
    IsActive?: boolean;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class LiveSessionRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'MeetingId';
    static readonly localTextPrefix = 'LiveSessions.LiveSession';
    static readonly lookupKey = 'LiveSessions.LiveSession';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<LiveSessionRow>('LiveSessions.LiveSession') }
    static async getLookupAsync() { return getLookupAsync<LiveSessionRow>('LiveSessions.LiveSession') }

    static readonly deletePermission = 'LiveSessions:LiveSessionsManagement:Modify';
    static readonly insertPermission = 'LiveSessions:LiveSessionsManagement:Modify';
    static readonly readPermission = 'LiveSessions:LiveSessionsManagement:View';
    static readonly updatePermission = 'LiveSessions:LiveSessionsManagement:Modify';

    static readonly Fields = fieldsProxy<LiveSessionRow>();
}